reducers/plugins/index.js

```diff
@@ -5,6 +5,44 @@ import {combineReducers} from 'redux';
 
 import {ActionTypes} from 'utils/constants.jsx';
 
+function addMenuItem(menu, data) {
+    if (menu.id === data.parentId) {
+        menu.subMenu = [...menu.subMenu, data];
+    } else if (menu.subMenu) {
+        menu.subMenu = menu.subMenu.map((item) => {
+            const subMenu = {...item};
+            return addMenuItem(subMenu, data);
+        });
+    }
+    return menu;
+}
+
+function buildSubMenu(state, action) {
+    const nextState = {...state};
+    const currentArray = nextState[action.name];
+    if (!currentArray) {
+        return action;
```

Comment:
can we return `null` here and have the caller just return the old state if it sees a `null` value. Otherwise believe it'll be hard to follow the logic.

Reply:
done.